src/service/tar_git

```diff
@@ -709,18 +712,48 @@ cache_prefetch () {
 handle_submodules() {
  
   if [ -f .gitmodules ] ; then
-    local submod url baseurl path URL REFERENCE
+    local submod url url_host url_proto url_path baseurl path URL REFERENCE
     baseurl=${1%/}
     for submod in $(git config -f .gitmodules -l | grep -P 'submodule\..*\.url' | xargs); do
       url=$(echo $submod | cut -d= -f2)
-      # Handle relative URLs
+      name=$(echo $submod | cut -d= -f1 | cut -d. -f2)
+      # Handle relative URLs and optionally configure local submodules
       case "$url" in
       ../*)
         url=${baseurl%/*}${url:2};;
       ./*)
         url=${baseurl}${url:1};;
+      http://* | https://* | ssh://* | git://* )
+        if [ "$LOCAL_SUBMODULES" == "Y" ] ; then
+          parent_url=$(git config --get remote.origin.url)
+          url_host=$(echo $parent_url | cut -d'/' -f3)
+          url_proto=$(echo $parent_url | cut -d'/' -f-2)
+          if echo "$LOCAL_GROUPS $LOCAL_MIRROR_GROUP" | tr ' ' '\n' | grep -q "^$(echo $url | cut -d'/' -f4)$"
```

Comment:
It isn't at all obvious what this is trying to do - I think some comments would be good.